src/conn/conn_api.c

```diff
@@ -1853,16 +1876,21 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
     }
 
     /*
-     * We own the database home, figure out if we're creating it. There are a few files created when
-     * initializing the database home and we could crash in-between any of them, so there's no
-     * simple test. The last thing we do during initialization is rename a turtle file into place,
-     * and there's never a database home after that point without a turtle file. If the turtle file
-     * doesn't exist, it's a create.
+     * If WiredTiger file exists but is size zero when it is not supposed to be (the turtle file
+     * exists and we are not salvaging), write a message but don't fail.
      */
-    WT_ERR(__wt_turtle_exists(session, &exist));
-    conn->is_new = exist ? 0 : 1;
+    empty = false;
+    if (fh) {
+        WT_ERR(__wt_filesize(session, fh, &size));
+        empty = (size_t)size == 0;
+        if (!is_salvage && !conn->is_new && empty)
+            WT_IGNORE_RET(__wt_msg(session, "WiredTiger version file is empty"));
```

Comment:
I agree that in an ideal world we would return an error here.  But because current WT releases will create an empty version file if it's not present, we have no idea how many existing customers might be in this situation.  Hopefully there aren't many.  But there could be a lot, and it would be ugly to have a lot of customers see a failure when they try to upgrade. 

So reporting the error tells customers there is something wrong without (I hope) generating a lot of HELP tickets and other fallout.

I discussed this a bit on [the ticket](https://jira.mongodb.org/browse/WT-7922?focusedCommentId=4274088&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4274088).